java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -456,6 +457,8 @@ private <ResponseT, ErrorT> ResponseT parseResponse(
                     ErrorT error = errorDeserializer.deserialize(parser, mapper);
                     throw new OpenSearchException((ErrorResponse) error);
                 }
+            } catch (OpenSearchException e) {
```

Comment:
Would be good to cover that with tests ... :-)

Reply:
Added integration test.